ws_api/wealthsimple_api.py

```
@@ -244,10 +178,24 @@ def check_oauth_token(
+                headers["x-ws-session-id"] = self.session.session_id
+            if self.session.wssdi:
+                headers["x-ws-device-id"] = self.session.wssdi
+            if WealthsimpleAPI.user_agent:
+                headers["User-Agent"] = WealthsimpleAPI.user_agent
+
+            try:
+                response = self.http.post(
+                    f"{self.OAUTH_BASE_URL}/token", json=data, headers=headers
+                ).json()
+            except RequestException as e:
+                raise CurlException(f"HTTP request failed: {e}")
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)